### PR TITLE
#140 :sparkles: support in project virtual environments

### PR DESCRIPTION
**Pull request by d-ryan-ashcraft**

### CODE REVIEW

File: habushu-maven-plugin/src/main/resources/META-INF/plexus/components.xml [hunk 1]
```diff
@@ -14,8 +14,8 @@
                         <id>default</id>
                         <phases>
                             <initialize>
```
Comment (d-ryan-ashcraft) [inline habushu-maven-plugin/src/main/resources/META-INF/plexus/components.xml:16]: I: just cleaned up the order here as we should make sure poetry is set up before initializing it

File: .mvn/extensions.xml [hunk 2]
```diff
@@ -2,6 +2,6 @@
     <extension>
         <groupId>org.apache.maven.extensions</groupId>
         <artifactId>maven-build-cache-extension</artifactId>
```
Comment (d-ryan-ashcraft) [inline .mvn/extensions.xml:4]: I (Informational): just bringing this up to date

**Review by jocobtt — APPROVED** [review]
Looks good to me!

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/CleanHabushuMojo.java [hunk 3]
```diff
@@ -107,42 +119,64 @@ public void execute() throws MojoExecutionException {
...
+        File venv = new File(inVirtualEnvironmentPath);
+
+        if (this.useInProjectVirtualEnvironment) {
+            getLog().debug("`in-project` virtual environment configured for this project");
+            if (StringUtils.isNotBlank(virtualEnvFullPath) && !inVirtualEnvironmentPath.equals(virtualEnvFullPath)) {
+                getLog().warn("'in-project' virtual environment is configured, but an external virtual environment was found!");
+                getLog().warn("Deleting external virtual environment: " + virtualEnvFullPath);
+                deleteVirtualEnv = true;
             }
+        } else {
```
Comment (jacksondelametter) [inline habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/CleanHabushuMojo.java:138]: Could this just be an else if?

Comment (d-ryan-ashcraft) [reply habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/CleanHabushuMojo.java:138]: It definitely could be.  I moved it to just try to make the logic a bit more straightforward to read.

**Pull request merged**